feat(bridge): enhance bridging with network fee estimation and UI improvements

amitabh94 opened the pull request:
## **Description**

This PR enhances the Bridge functionality with improved gas fee estimation, UI refinements, and dependency updates. The changes focus on providing more accurate network fee calculations and a better user experience.

Key improvements:

- Implemented `useGasFeeEstimates` hook with comprehensive test coverage
- Enhanced `useBridgeQuoteData` with improved network fee formatting
- Integrated gas fee estimates into BridgeView for accurate fee calculations
- Added `useUnifiedSwapBridgeContext` hook for better state management
- Updated UI components for better user experience
- Updated bridge controller dependencies to latest versions

## **Related issues**

Fixes: [MMS-2265](https://consensyssoftware.atlassian.net/browse/MMS-2265)
Fixes: [MMS-2300](https://consensyssoftware.atlassian.net/browse/MMS-2300)
Fixes: [MMS-2301](https://consensyssoftware.atlassian.net/browse/MMS-2301)

## **Manual testing steps**

1. Navigate to the Bridge feature
2. Verify gas fee estimates are displayed correctly
3. Test the QuoteInfoModal and SlippageModal UI changes
4. Verify keyboard display handling in UI components
5. Test network fee calculations in bridge transactions

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/838b1f7f-3b76-422b-9998-7e573f9a3b3e

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

[MMS-2265]: https://consensyssoftware.atlassian.net/browse/MMS-2265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MMS-2300]: https://consensyssoftware.atlassian.net/browse/MMS-2300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MMS-2301]: https://consensyssoftware.atlassian.net/browse/MMS-2301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

amitabh94 committed:
fix: update network fee handling in useBridgeQuoteData hook

- Refactored the network fee assignment to use the value from activeQuote instead of a hardcoded string.

amitabh94 committed:
chore: update changelog

amitabh94 commented:
@micaelae @GustavoRSSilva I am getting empty values for `totalNetworkFee` from the quote from the controller selectors. Would you know why this could be ?

amitabh94 committed:
fix: update useBridgeQuoteData tests for total network fee

micaelae commented:
@amitabh94 are you getting gasFee metadata values? I ran into this today and opened a PR for core (it's still open). Can you try with that preview build?

amitabh94 commented:
> @amitabh94 are you getting gasFee metadata values? I ran into this today and opened a PR for core (it's still open). Can you try with that preview build?

Can you please share the link of the core PR ?

amitabh94 committed:
fix: update bottom content when keyboard is displayed

amitabh94 committed:
fix: update condition to check for input focus instead of keypad display for rendering logic

amitabh94 committed:
chore: update bridge-controller and bridge-status-controller dependencies to latest versions

amitabh94 committed:
fix: implement placeholder for trackMetaMetricsFn in Engine class

amitabh94 committed:
fix: update import path for StatusTypes in initialState mock

amitabh94 committed:
fix: update import paths for StatusTypes and BridgeHistoryItem in transaction details components

amitabh94 committed:
chore:  improve test formatting

amitabh94 committed:
fix: enhance BridgeView tests and initial state mock

- Added mocks for GasFeeController and NetworkController in BridgeView tests.
- Updated BridgeController mock to include updateBridgeQuoteRequestParams.
- Mocked Skeleton component to prevent animation during tests.
- Introduced TransactionController in initial state mock with an empty transactions array.

amitabh94 committed:
feat: add useGasFeeEstimates hook and corresponding tests

- Implemented the useGasFeeEstimates hook to retrieve and poll gas fee estimates based on the network client ID.
- Created unit tests for useGasFeeEstimates to validate polling behavior and gas fee estimates retrieval.
- Mocked necessary dependencies including GasFeeController and NetworkController for testing purposes.

amitabh94 committed:
feat: implement useUnifiedSwapBridgeContext hook and integrate with useBridgeQuoteRequest

amitabh94 committed:
feat: integrate gas fee estimates  into BridgeView

- Added useGasFeeEstimates hook to fetch gas fee estimates based on the selected network client ID.
- Updated BridgeView to utilize the gas fee estimates for improved transaction processing.

amitabh94 committed:
feat: enhance useBridgeQuoteData hook with network fee formatting

- Refactored useBridgeQuoteData to include a new getNetworkFee function for calculating and formatting network fees based on the selected primary currency.
- Updated tests to validate network fee formatting for both ETH and USD currencies.
- Improved overall code structure by utilizing useCallback for memoization and ensuring proper dependencies in useMemo.

amitabh94 committed:
feat: add marginBottom to QuoteDetailsCard styles

- Updated QuoteDetailsCard styles to include a marginBottom of 12 for improved layout consistency.
- Adjusted snapshots to reflect the new marginBottom property in the rendered output.

amitabh94 committed:
feat: update SlippageModal styles and snapshots

- Added a new `segmentedControl` style with a gap of 8 for improved layout in the SlippageModal component.
- Adjusted the gap in the SlippageModal snapshot to reflect the updated styling.
- Ensured consistency in the UI presentation of slippage options.

amitabh94 committed:
feat: refactor QuoteInfoModal to remove footer button implementation

- Removed the footer button props and related components from QuoteInfoModal for future implementation.
- Added TODO comments to indicate the planned reintroduction of footer functionality for viewing other quotes.
- Updated snapshots to reflect the changes in the QuoteInfoModal component structure.

amitabh94 committed:
feat: refactor useBridgeQuoteData to utilize ticker instead of native currency

- Updated the useBridgeQuoteData hook to replace the native currency selection with ticker selection for improved consistency in quote formatting.
- Adjusted the formatted amount to display the ticker instead of the native currency.
- Modified dependencies in useMemo to reflect the change from nativeCurrency to ticker.

amitabh94 committed:
Merge branch 'main' into MMS-2265--display-real-network-fee-for-bridge

amitabh94 committed:
chore: update changelog

amitabh94 committed:
feat: remove common useGasFeeEstimates hook and associated tests in lieu of confirmation gas hooks

amitabh94 commented on app/core/Engine/Engine.ts:
```diff
@@ -1077,6 +1077,9 @@ export class Engine {
       config: {
         customBridgeApiBaseUrl: BRIDGE_DEV_API_BASE_URL,
       },
+      trackMetaMetricsFn: () => {
```
@bfullam This is what we need to implement. You can look at @micaelae ' s Extension PR to see how to integrate this.

amitabh94 committed:
feat: enhance useBridgeQuoteData with mock data and network fee handling

- Introduced a new mockQuoteWithMetadata for testing purposes to standardize quote data.
- Updated the useBridgeQuoteData hook to handle cases where totalNetworkFee may be undefined or missing, ensuring robust error handling.
- Enhanced unit tests to cover various scenarios for network fee formatting, including cases with missing data and default currency handling.
- Improved overall test coverage for the useBridgeQuoteData hook by integrating the new mock data.
